assets/js/booru.js

```diff
@@ -21,11 +34,31 @@ function clearTags() {
   });
 }
 
-/* Returns a single tag, or a dummy tag object if we don't know about it yet */
+/**
+ * @param {unknown} value
+ * @returns {value is TagData}
+ */
+function isValidStoredTag(value) {
```

Comment:
This feels a bit silly... if we only ever store valid tags here, why test against other possibilities?

Reply:
This is a future TypeScript type guard, it will be more useful later